src/dawn/CodeGen/Cuda/CudaCodeGen.cpp

```diff
@@ -581,7 +615,13 @@ std::string CudaCodeGen::generateStencilInstantiation(
 
   Namespace cudaNamespace("cuda", ssSW);
 
-  generateAllCudaKernels(ssSW, stencilInstantiation);
+  // map from MS ID to cacheProperty
+  std::unordered_map<int, CacheProperties> cachePropertyMap;
```

Comment:
why is this locally defined rather than a private class member? now we're passing it to basically every fctn

Reply:
yeah, that is a good point, I ll change it

Reply:
done